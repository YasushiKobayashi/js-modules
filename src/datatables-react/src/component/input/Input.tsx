import * as React from 'react'
import styled from 'styled-components'

interface Props {
  label?: string;
  value: string;
  handleInput: (v: string) => void;
}

const id = 'react-datatable-input'

const Wrapper = styled('div')`
  display: flex;

  input {
    width: 100%;
  }
`

const Input: React.FC<Props> = props => {
  const { label, value, handleInput } = props

  const labelContent = label ? <label htmlFor={id}>{label}</label> : null

  return (
    <Wrapper>
      {labelContent}
      <input
        type="text"
        id={id}
        data-testid={id}
        className={id}
        onChange={e => handleInput(e.currentTarget.value)}
        value={value}
      />
    </Wrapper>
  )
}

export { Input as default }
