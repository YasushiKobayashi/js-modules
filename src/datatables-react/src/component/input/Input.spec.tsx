import * as React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';

import Input from './Input';

const value = 'value';
const id = 'react-datatable-input';
const base = {
  value,
};

describe('Input', () => {
  afterEach(() => {
    cleanup();
  });

  it('handleInput called input change', () => {
    const handleInput = jest.fn();
    const props = Object.assign({}, base, { handleInput });
    const { getByTestId } = render(<Input {...props} />);

    const input = getByTestId(id) as HTMLInputElement;
    expect(input.value).toBe(value);

    fireEvent.change(input, { target: { value: id } });
    expect(handleInput).toBeCalled();
    expect(handleInput.mock.calls[0][0]).toBe(id);
  });

  it('renders correctly', () => {
    const handleInput = jest.fn();
    const props = Object.assign({}, base, { handleInput });
    const { asFragment } = render(<Input {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
