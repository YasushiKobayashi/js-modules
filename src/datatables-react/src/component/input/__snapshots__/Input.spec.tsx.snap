// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm fKxEoa"
  >
    <input
      class="react-datatable-input"
      data-testid="react-datatable-input"
      id="react-datatable-input"
      type="text"
      value="value"
    />
  </div>
</DocumentFragment>
`;
